### ISSUE TITLE
Wifi 802.1x still NOT working on Cyannogen

### ISSUE THREAD

**Issue by lans38** [CyanogenDefy/android_device_motorola_jordan#73]
connect...disconnect...connect .... thats the result if I try to establish a connection to a 802.1x network.

We have several wifi networks here at my office. But 802.1x wifi networks with PEAP+GTC and wifi networks with PEAP without Phase-2-Authentification are NOT working. I don't know if it is a generally cyanogen issue. With Stock ROMs (Eclair,Froyo,Gingerbread) Wifi is working without any problems. I think it is is known at least since CM7 beta 2. 
That problem is still avoiding me to use Cyanogen on my Defy.

BTW: I can't  change any settings at our wifi APs!!!

**Comment by killigrew** [comment]
could you please at least post which version you are using, 
as nobody's complaining about this issue atm in the main thread on xda.

**Comment by lans38** [comment]
CM7 RC1v2 and also tried last nightly cm7-110726-0018-BOOTSEA-Jordan

**Comment by ghost** [comment]
not working here either.. big issue!
already mentioned the problem in version beta6

**Comment by killigrew** [comment]
can you please be a bit more descriptive about the problem?
what does "not working" exactly mean?
can you provide i.e. some logcat outputs or anything?

greetings

**Comment by ghost** [comment]
sorry, its just because there is not much to describe
if you want to connect to an access point wich uses 802.1x protocol for authentication.. it does not connect, as described in the first post
i'll try to post a logcat file soon

**Comment by ghost** [comment]
I(16572) Trying to associate with 00:15:fa:b3:27:c0 (SSID='802.1X' freq=2412 MHz)  (wpa_supplicant)
I(16572) CTRL-EVENT-STATE-CHANGE id=-1 state=3 BSSID=00:15:fa:b3:27:c0  (wpa_supplicant)
I(16572) CTRL-EVENT-STATE-CHANGE id=6 state=4 BSSID=00:15:fa:b3:27:c0  (wpa_supplicant)
I(16572) Associated with 00:15:fa:b3:27:c0  (wpa_supplicant)
I(16572) CTRL-EVENT-EAP-STARTED EAP authentication started  (wpa_supplicant)
I(16572) CTRL-EVENT-EAP-FAILURE EAP authentication failed  (wpa_supplicant)
I(16572) Authentication with 00:15:fa:b3:27:c0 timed out.  (wpa_supplicant)
I(16572) CTRL-EVENT-STATE-CHANGE id=6 state=0 BSSID=00:00:00:00:00:00  (wpa_supplicant)
I(16572) CTRL-EVENT-STATE-CHANGE id=-1 state=2 BSSID=00:00:00:00:00:00  (wpa_supplicant)
E(16572) ERROR - wpa_driver_tista_private_send - error sending Wext private IOCTL to STA driver (ioctl_cmd = 8021501,  res = -1, errno = 95)  (wpa_supplicant)
E(16572) ERROR - Failed to do tista scan!  (wpa_supplicant)
W(16572) Failed to initiate AP scan.  (wpa_supplicant)
I(16572) CTRL-EVENT-DISCONNECTED - Disconnect event - remove keys  (wpa_supplicant)
I(16572) CTRL-EVENT-STATE-CHANGE id=-1 state=0 BSSID=00:00:00:00:00:00  (wpa_supplicant)

this is all i found and figured relevant, so for some reason the authentication fails..
passwords etc. are 100% correct

**Comment by ghost** [comment]
well seems i had some problems selecting the correct method here at university
if followed this instruction: http://android.modaco.com/topic/312366-securew2-ttls-alternative-for-htc-hero/ (2nd post)

have to try it at my home network and see if i can solve the problem there too
as for my university i had to select TTLS and PAP

im actually surprised it worked

**Comment by killigrew** [comment]
ah ok, i'm already using TTLS for my uni wlan, but I can't login via PEAP on my Linux Notebook and
never could with the stock firmware of the defy, too

**Comment by ghost** [comment]
well.. still cant get it to work at home with my PEAP/MSCHAPV2 configuration
the EAP-TTLS config at university works fine

but maybe this is an cyanogenmod issue
as you can read here: http://code.google.com/p/cyanogenmod/issues/detail?id=3473

and with the stock rom.. my configuration was working fine also

**Comment by lans38** [comment]
here a logcat when I try to connect with a CM7 flashed Defy: 
(I could also submit some logging information of a successful connection by a "Gingerbread ROM 2.3.4/4.5.1-134" based  Defy next week if it is needed.)

07/29/2011 09:50:29 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=-1 state=2 BSSID=00:00:00:00:00:00
07/29/2011 09:50:29 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=-1 state=2 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:29 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: SCANNING ==> SCANNING
07/29/2011 09:50:32 [INFORMATION] wpa_supplicant(2168) Trying to associate with 000b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)
07/29/2011 09:50:32 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=-1 state=3 BSSID=000b:85:11:11:11
07/29/2011 09:50:32 [VERBOSE] WifiMonitor(2101) Event [Trying to associate with 000b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)]
07/29/2011 09:50:32 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=-1 state=3 BSSID=000b:85:11:11:11]
07/29/2011 09:50:32 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: SCANNING ==> ASSOCIATING
07/29/2011 09:50:32 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=4 BSSID=000b:85:11:11:11
07/29/2011 09:50:32 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=4 BSSID=000b:85:11:11:11]
07/29/2011 09:50:32 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: ASSOCIATING ==> ASSOCIATED
07/29/2011 09:50:32 [INFORMATION] wpa_supplicant(2168) Associated with 000b:85:11:11:11
07/29/2011 09:50:32 [VERBOSE] WifiMonitor(2101) Event [Associated with 000b:85:11:11:11]
07/29/2011 09:50:32 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-EAP-STARTED EAP authentication started
07/29/2011 09:50:32 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-EAP-STARTED EAP authentication started]
07/29/2011 09:50:33 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-EAP-METHOD EAP vendor 0 method 25 (PEAP) selected
07/29/2011 09:50:33 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-EAP-METHOD EAP vendor 0 method 25 (PEAP) selected]
07/29/2011 09:50:33 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-EAP-SUCCESS EAP authentication completed successfully
07/29/2011 09:50:33 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-EAP-SUCCESS EAP authentication completed successfully]
07/29/2011 09:50:33 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00
07/29/2011 09:50:33 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:33 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: ASSOCIATED ==> FOUR_WAY_HANDSHAKE
07/29/2011 09:50:33 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-DISCONNECTED - Disconnect event - remove keys
07/29/2011 09:50:33 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-DISCONNECTED - Disconnect event - remove keys]
07/29/2011 09:50:33 [VERBOSE] WifiStateTracker(2101) New network state is DISCONNECTED
07/29/2011 09:50:33 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=0 BSSID=00:00:00:00:00:00
07/29/2011 09:50:33 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=0 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:33 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: FOUR_WAY_HANDSHAKE ==> DISCONNECTED
07/29/2011 09:50:33 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=2 BSSID=00:00:00:00:00:00
07/29/2011 09:50:33 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=2 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:33 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: DISCONNECTED ==> SCANNING
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) Trying to associate with 000b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=3 BSSID=000b:85:11:11:11
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [Trying to associate with 000b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)]
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=3 BSSID=000b:85:11:11:11]
07/29/2011 09:50:34 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: SCANNING ==> ASSOCIATING
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=4 BSSID=000b:85:11:11:11
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=4 BSSID=000b:85:11:11:11]
07/29/2011 09:50:34 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: ASSOCIATING ==> ASSOCIATED
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) Associated with 000b:85:11:11:11
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [Associated with 000b:85:11:11:11]
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-DISCONNECTED - Disconnect event - remove keys
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-DISCONNECTED - Disconnect event - remove keys]
07/29/2011 09:50:34 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=0 BSSID=00:00:00:00:00:00
07/29/2011 09:50:34 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=0 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:34 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: ASSOCIATED ==> FOUR_WAY_HANDSHAKE
07/29/2011 09:50:34 [VERBOSE] WifiStateTracker(2101) New network state is DISCONNECTED
07/29/2011 09:50:34 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: FOUR_WAY_HANDSHAKE ==> DISCONNECTED
07/29/2011 09:50:35 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=2 BSSID=00:00:00:00:00:00
07/29/2011 09:50:35 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=2 BSSID=00:00:00:00:00:00]
07/29/2011 09:50:35 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: DISCONNECTED ==> SCANNING
07/29/2011 09:50:36 [INFORMATION] wpa_supplicant(2168) Trying to associate with 000b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)
07/29/2011 09:50:36 [INFORMATION] wpa_supplicant(2168) CTRL-EVENT-STATE-CHANGE id=1 state=3 BSSID=000b:85:11:11:11
07/29/2011 09:50:36 [VERBOSE] WifiMonitor(2101) Event [Trying to associate with 000b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)]
07/29/2011 09:50:36 [VERBOSE] WifiMonitor(2101) Event [CTRL-EVENT-STATE-CHANGE id=1 state=3 BSSID=000b:85:11:11:11]
07/29/2011 09:50:36 [VERBOSE] WifiStateTracker(2101) Changing supplicant state: SCANNING ==> ASSOCIATING
07/29/2011 09:50:37 [ERROR] wpa_supplicant(2168) Ongoing Scan action...
07/29/2011 09:50:38 [DEBUG] WifiStateTracker(2101) Reset connections and stopping DHCP
07/29/2011 09:50:39 [DEBUG] WifiStateTracker(2101) Disabling interface

**Comment by lans38** [comment]
logcat from a successful wifi 802.1x connection (Done by a Defy with Gingerbread ROM 2.3.4/4.5.1-134)

If you compare both logs you can see that CM7 fails on the "FOUR_WAY_HANDSHAKE". 

8/01/2011 09:34:54 [INFORMATION] wpa_supplicant(1458) Trying to associate with 00:0b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)
08/01/2011 09:34:54 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-STATE-CHANGE id=-1 state=3 BSSID=00:0b:85:11:11:11
08/01/2011 09:34:54 [VERBOSE] WifiMonitor(1377) Event [Trying to associate with 00:0b:85:11:11:11 (SSID='WLAN - 802.1x (PEAP-GTC)' freq=2412 MHz)]
08/01/2011 09:34:54 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-STATE-CHANGE id=-1 state=3 BSSID=00:0b:85:11:11:11]
08/01/2011 09:34:55 [VERBOSE] WifiStateTracker(1377) Changing supplicant state: SCANNING ==> ASSOCIATING
08/01/2011 09:34:55 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-STATE-CHANGE id=1 state=4 BSSID=00:0b:85:11:11:11
08/01/2011 09:34:55 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-STATE-CHANGE id=1 state=4 BSSID=00:0b:85:11:11:11]
08/01/2011 09:34:55 [VERBOSE] WifiStateTracker(1377) Changing supplicant state: ASSOCIATING ==> ASSOCIATED
08/01/2011 09:34:55 [INFORMATION] wpa_supplicant(1458) Associated with 00:0b:85:11:11:11
08/01/2011 09:34:55 [VERBOSE] WifiMonitor(1377) Event [Associated with 00:0b:85:11:11:11]
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-EAP-STARTED EAP authentication started
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-EAP-STARTED EAP authentication started]
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-EAP-METHOD EAP vendor 0 method 25 (PEAP) selected
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-EAP-METHOD EAP vendor 0 method 25 (PEAP) selected]
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-EAP-SUCCESS EAP authentication completed successfully
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-EAP-SUCCESS EAP authentication completed successfully]
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00]
08/01/2011 09:34:56 [VERBOSE] WifiStateTracker(1377) Changing supplicant state: ASSOCIATED ==> FOUR_WAY_HANDSHAKE
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-STATE-CHANGE id=1 state=5 BSSID=00:00:00:00:00:00]
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-STATE-CHANGE id=1 state=6 BSSID=00:00:00:00:00:00
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) WPA: Key negotiation completed with 00:0b:85:11:11:11 [PTK=CCMP GTK=TKIP]
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-STATE-CHANGE id=1 state=7 BSSID=00:00:00:00:00:00
08/01/2011 09:34:56 [INFORMATION] wpa_supplicant(1458) CTRL-EVENT-CONNECTED - Connection to 00:0b:85:11:11:11 completed (reauth) [id=1 id_str=]
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-STATE-CHANGE id=1 state=6 BSSID=00:00:00:00:00:00]
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [WPA: Key negotiation completed with 00:0b:85:11:11:11 [PTK=CCMP GTK=TKIP]]
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-STATE-CHANGE id=1 state=7 BSSID=00:00:00:00:00:00]
08/01/2011 09:34:56 [VERBOSE] WifiMonitor(1377) Event [CTRL-EVENT-CONNECTED - Connection to 00:0b:85:11:11:11 completed (reauth) [id=1 id_str=]]
08/01/2011 09:34:56 [VERBOSE] WifiStateTracker(1377) Changing supplicant state: FOUR_WAY_HANDSHAKE ==> FOUR_WAY_HANDSHAKE
08/01/2011 09:34:56 [VERBOSE] WifiStateTracker(1377) Changing supplicant state: FOUR_WAY_HANDSHAKE ==> GROUP_HANDSHAKE
08/01/2011 09:34:56 [VERBOSE] WifiStateTracker(1377) Changing supplicant state: GROUP_HANDSHAKE ==> COMPLETED
08/01/2011 09:34:56 [VERBOSE] WifiStateTracker(1377) New network state is CONNECTED

**Comment by ghost** [comment]
any news here? installed nightly from 0809.. problem still persists

**Comment by lans38** [comment]
nightly from 0811...no changes. still not working.

**Comment by claria** [comment]
Hi,
think this bug is related to wpa_supplicant. I replaced wpa_supplicant with  the one from gingerbread rom. 

I uploaded the files from gingebread rom. 
Only replace wpa_supplicant and dont forget to change the permissions to rwxr-xr-x.
http://dl.dropbox.com/u/283099/gingerwpa.zip

**Comment by ghost** [comment]
tried it with your attached files, just replaced the wpa_supplicant, changed permissions and owner
still not working!

**Comment by lans38** [comment]
not working for me too.

**Comment by claria** [comment]
hmm could you try replacing wlan_loader and wlan_cu as well. i did that and it started working for me
http://dl.dropbox.com/u/283099/ms2ginger_wpa_supplicant.zip

**Comment by lans38** [comment]
So I got a successful connection. But with different files as in last post. I take the wpa_supplicant from the xda post (see link below). 
After replacing the file and change the permissions (all done with adb) I restarted the phone and wiped the cache and the dalvik cache in the custom recovery. Now wifi 802.1x with PEAP+MSchapv2 is working now for me. 

http://forum.xda-developers.com/showpost.php?p=11356793&postcount=391
